### PR TITLE
PLU-165: Refactor step error

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem

Currently, there are helper methods which return new `StepError` and adds another layer of complexity and does not conform to error throwing conventions.

## Solution

Remove the helper functions and throw new `StepError` directly instead at the point of error.

## Tests
- Checked that errors are still being displayed for changes made to the various apps involved e.g. postman, telegram

### CODE REVIEW

File: packages/backend/src/apps/custom-api/actions/http-request/index.ts
Comment (liangyuanruo) [inline packages/backend/src/apps/custom-api/actions/http-request/index.ts]: Can just in-line the variable if it's only used once

File: packages/backend/src/apps/postman/common/throw-errors.ts
Comment (liangyuanruo) [inline packages/backend/src/apps/postman/common/throw-errors.ts:19]: Would having a sub-error type help, instead of parsing the error message and also having to ensure that it doesn't clash with any other error messages?

File: packages/backend/src/errors/step.ts
Comment (liangyuanruo) [inline packages/backend/src/errors/step.ts:18]: nit: is there any way to avoid this typecast? Generally we try to avoid doing this unless absolutely necessary

**Review by liangyuanruo — COMMENTED** [review]
Thanks for the tag - I did a cursory review. Only minor issues; will leave it to @pregnantboy to approve since he's back.

File: packages/backend/src/apps/custom-api/actions/http-request/index.ts
Comment (m0nggh) [reply packages/backend/src/apps/custom-api/actions/http-request/index.ts]: okay, thanks!

File: packages/backend/src/apps/postman/common/throw-errors.ts
Comment (m0nggh) [reply packages/backend/src/apps/postman/common/throw-errors.ts:19]: I thought that we discussed that sub-error types to just return the specific postman error e.g. `BlacklistedPostmanError` doesn't rly add value if it just returns the error message string. And can I ask how would having a sub-error type help to avoid parsing the error message here cus I'm not sure. Thanks!

File: packages/backend/src/errors/step.ts
Comment (m0nggh) [reply packages/backend/src/errors/step.ts:18]: I think the `details` is a `Record<string, unknown>` in the `base.ts` file, which is a superset of the type `IJSONObject`, not sure how else to avoid this other than to just amend the type in the `base.ts` file instead

Comment (liangyuanruo) [reply packages/backend/src/errors/step.ts:18]: OK. I think we leave any refactoring around this out for now.

File: packages/backend/src/apps/postman/common/throw-errors.ts
Comment (liangyuanruo) [inline packages/backend/src/apps/postman/common/throw-errors.ts]:
```suggestion
  const { code: errorCode, message: errorMessage } = postmanErrorData
```

File: packages/backend/src/apps/postman/common/throw-errors.ts
Comment (liangyuanruo) [reply packages/backend/src/apps/postman/common/throw-errors.ts:19]: I see, I mis-interpreted the `errorMessage` variable here. My mistake, no worries!

**Review by liangyuanruo — APPROVED** [review]
Thanks @m0nggh ! Can merge after addressing the remaining comments.

**Comment by m0nggh** [comment]
thanks for reviewing @liangyuanruo, will just get @pregnantboy to also take a look to align this new refactor and make sure all is good before merging!

**Pull request merged**
